editor/scene_tree_editor.cpp

```diff
@@ -645,6 +645,13 @@ void SceneTreeEditor::_rename_node(ObjectID p_node, const String &p_name) {
 	TreeItem *item = _find(tree->get_root(), n->get_path());
 	ERR_FAIL_COND(!item);
 
+	// Nodes must have a non empty name, so resets it to previous text and show warning
+	if (p_name.empty()) {
```

Comment:
You should check `p_name.strip_edges().empty()`, spaces only are also illegal.

Reply:
Right, I forgot.